app/gui2/src/providers/graphNavigator.ts

```diff
@@ -0,0 +1,6 @@
+import { useNavigator } from '@/util/navigator'
+import { createProvidable } from '.'
+
+export type GraphSelection = ReturnType<typeof useNavigator>
+const { provideFn, useFn } = createProvidable(useNavigator)
+export { provideFn as provideGraphNavigator, useFn as useGraphNavigator }
```

Comment:
I bit confused of the meanings of `use`. The provider suggests that if some function is called `use` that means that we use something provided elsewhere. But `useNavigator` does not have corresponding `provideNavigator`. For clarity, I would reserve `use` prefix for something what is actually _provided_ (so rename the useNavigator in this case).

Reply:
Also, my impression is that this `createProvidable` creates quite a thin wrappers for something like just `provide(NavigatorComposable, useNavigator(args...))`. We could just write that in every "provide" usage place, and it will be more clear IMO (one layer of abstraction less to comprehend).

Reply:
It is very similar to `createStore` function from pinia. The abstraction is very thin, mostly for removing the repeated boilerplate, and safeguarding against missing context values. Therefore making the code easier to read after all. It just needs some docs.

I'm going to rename `use` to `inject`, to signify that this is indeed different from traditional vue composables (i.e. it expects a `provide` method to be called by any of it parents).